src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -112,6 +112,10 @@ class FastSync(
     private var requestedBlockBodies: Map[ActorRef, Seq[ByteString]] = Map.empty
     private var requestedReceipts: Map[ActorRef, Seq[ByteString]] = Map.empty
 
+
+    private val MAXIMUM_CACHE_SIZE = 500 // todo adjust cache size
+    private val blockSyncCache = CacheBuilder.newBuilder().maximumSize(MAXIMUM_CACHE_SIZE).build[ByteString, (Option[BlockHeader], Boolean, Boolean, Option[BigInt])]()
```

Comment:
I think it would be clearer to define some case class for this tuple `(Option[BlockHeader], Boolean, Boolean, Option[BigInt]`  which is cache value .